src/components/AutoCompleteInput.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverBody,
+  useColorModeValue,
+} from "@chakra-ui/react";
+import { Box, Text } from "@chakra-ui/react";
+import React, { useCallback, useEffect } from "react";
+import { useRef, useState } from "react";
+
+type AutoCompleteProps = {
+  children: React.ReactNode;
+  triggerRef: React.RefObject<HTMLTextAreaElement>;
+  parentRef: React.RefObject<HTMLDivElement>;
+  availablePrompts: {
+    command: string;
+    helpTitle: string;
+    helpDescription: string;
+  }[];
+};
+
+const AutoCompleteInput = ({
+  children,
+  triggerRef,
+  parentRef,
+  availablePrompts,
+}: AutoCompleteProps) => {
+  const [popupPosition, setPopupPosition] = useState<number>(0);
+  const [inputWidth, setInputWidth] = useState<number>(0);
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [selectedIndex, setSelectedIndex] = useState<number>(-1);
```

Comment:
```suggestion
  const [focusedIndex, setFocusedIndex] = useState<number>(-1);
```

`focusedIndex` would be a better name since this represents the focused item on the list.